File: task-flow-frontend/src/app/utils/task-status.ts
import { TaskStatus } from '../components/enum/task-status.enum';
import { TaskModel } from '../components/task/task';

export const separateTasksByStatus = (tasks: TaskModel[]) => {
  const todo: TaskModel[] = [];
  const inProgress: TaskModel[] = [];
  const review: TaskModel[] = [];
  const done: TaskModel[] = [];

  tasks.forEach((element: TaskModel) => {
    switch (element.status) {
      case TaskStatus.IN_PROGRESS:
        inProgress.push(element);
        break;
      case TaskStatus.REVIEW:
        review.push(element);
        break;
      case TaskStatus.DONE:
        done.push(element);
        break;
      default:
        todo.push(element);
    }
  });

  return { todo, inProgress, review, done };
};

File: src/main/java/com/max/TaskFlow/model/TaskStatus.java
package com.max.TaskFlow.model;

import com.fasterxml.jackson.annotation.JsonProperty;

public enum TaskStatus {
    @JsonProperty("To do") TO_DO,
    @JsonProperty("In progress") IN_PROGRESS,
    @JsonProperty("Review") REVIEW,
    @JsonProperty("Done") DONE
}